### PR TITLE
Adding Makefile

### PR DESCRIPTION
**Pull request by Fenikkusu**
Hello!

* Type: new feature

**In raising this pull request, I confirm the following:**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)
- [x] I have checked that another pull request for this purpose does not exist
- [ ] I wrote some tests for this PR
- [ ] I have updated the relevant CHANGELOG
- [ ] I have created a PR for the [documentation](https://github.com/phalcon/docs) about this change

Small description of change:

I feel like I may have submitted this in the past, but could be wrong. This may not be something everyone wants anyways. This is the Makefile I use to develop with. It makes it fairly simple to work with projects. It runs everything through docker, so I don't have to worry about installing any dependencies on my local computer.

`make build`
`make test-unit`

etc.

Thanks

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+######################################################
+# OS X / Linux                                       #
+#                                                    #
+# Likely Already Installed. Otherwise use the os's   #
+# package manager (IE: apt-get, brew, macports, etc) #
+######################################################
+
+CACHE_PATH := cache
+BIN_PATH := bin
+NODE_MODULES := node_modules
```
Comment (sergeyklay) [inline Makefile]: Why do we need Node here?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+PHP-CS-FIXER_URL := https://cs.sensiolabs.org/download/php-cs-fixer-v2.phar
```
Comment (sergeyklay) [inline Makefile]: You can safely remove this

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+# Binaries
+vendor/codeception/codeception/codecept: | vendor
+
+$(BIN_PATH)/.docker-login: | $(BIN_PATH)
+	docker login registry.gitlab.com
+	touch $@
+
+# Docker Installation
+$(BIN_PATH)/.docker: | $(BIN_PATH)
+	which docker || (echo "Docker Is Not Installed" && exit 1)
```
Comment (sergeyklay) [inline Makefile]:
```suggestion
	command -v docker 2>/dev/null || (echo "Docker Is Not Installed" && exit 1)
```

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+######################################################
+# Windows                                            #
+#                                                    #
+# http://gnuwin32.sourceforge.net/downlinks/make.php #
+######################################################
+# OS X / Linux                                       #
+#                                                    #
+# Likely Already Installed. Otherwise use the os's   #
+# package manager (IE: apt-get, brew, macports, etc) #
+######################################################
```
Comment (sergeyklay) [inline Makefile]: Could you add standard Phalcon's license note at the top

**Review by sergeyklay — CHANGES_REQUESTED** [review]
Looks good to me except some things

**Comment by CameronHall** [comment]
This is an awesome idea.

File: Makefile (hunk 4 above)
Comment (Fenikkusu) [reply Makefile]: Tracking: Done

File: Makefile (hunk 3 above)
Comment (Fenikkusu) [reply Makefile]: Tracking: Done

File: Makefile (hunk 2 above)
Comment (Fenikkusu) [reply Makefile]: Tracking: Done

File: Makefile (hunk 1 above)
Comment (Fenikkusu) [reply Makefile]: Tracking: Done

**Comment by Fenikkusu** [comment]
I've made the requested changes. I also did some minor cleanup to the Codecept Commands to simplify them.

**Review by ruudboon — APPROVED** [review]

**Review by sergeyklay — APPROVED** [review]

**Comment by sergeyklay** [comment]
Thank you

**Pull request merged**